src/upgrades/v0_9_0/ca_objects_migration.rs

```diff
@@ -103,33 +172,108 @@ impl UpgradeStore for CasStoreMigration {
+            let time_for_init_command = match cmd_keys.first() {
+                Some(first_command) => {
+                    let old_cmd: OldStoredCaCommand = self.get(&first_command)?;
+                    old_cmd.time
+                }
+                None => Time::now(),
+            };
+
+            // Migrate the init event.
+            let old_init_key = KeyStoreKey::scoped(migration_scope.clone(), "delta-0.json".to_string());
+            let init_key = KeyStoreKey::scoped(scope.clone(), "delta-0.json".to_string());
+
+            let old_init: OldCaIni = self.get(&old_init_key)?;
+            let (id, _, old_ini_det) = old_init.unpack();
+            let (rfc_8183_id, repo_opt, ta_opt) = old_ini_det.unpack();
+            let ini = IniDet::new(&id, rfc_8183_id.into());
+            debug!("  init: {}", init_key);
+            self.store.store(&init_key, &ini)?;
+
+            // If the CA was initialized with an embedded repository, then make sure that
+            // we generate a command + events to update the repository to the 'local' RFC 8181
+            // version of this repository.
+
+            if repo_opt.is_some() {
+                debug!("  +- Generate command and event for initialized repository");
+                let res = self
+                    .repo_manager
+                    .as_ref()
+                    .as_ref()
+                    .ok_or(UpgradeError::KrillError(
+                        crate::commons::error::Error::RepositoryServerNotEnabled,
+                    ))?
+                    .repository_response(&id)?;
+
+                let contact = RepositoryContact::new(res);
+                let service_uri = contact.service_uri().clone();
+
+                info.last_event += 1;
+                let event = CaEvt::new(&id, info.last_event, CaEvtDet::RepoUpdated { contact });
+
+                let command = StoredCaCommand::new(
+                    dflt_actor.clone(),
+                    time_for_init_command,
+                    id,
+                    info.last_event,
+                    info.last_command,
+                    StorableCaCommand::RepoUpdate { service_uri },
+                    StoredEffect::Success { events: vec![1] },
+                );
+
+                info.last_command += 1;
+                info.last_update = time_for_init_command;
+
+                let event_key = Self::event_key(&scope, info.last_event);
+                self.store.store(&event_key, &event)?;
+
+                let cmd_key = CommandKey::for_stored(&command);
+                let key = KeyStoreKey::scoped(scope.clone(), format!("{}.json", cmd_key));
+
+                self.store.store(&key, &command)?;
+            }
+
+            // If the CA was initialized as a trust anchor, then we refuse to upgrade.
+            // This can only happen if this is a very old test system. People will need
+            // to set up a new test system instead.
+            if ta_opt.is_some() {
```

Comment:
Should this check be above the if block that stores RFC 8181 migration events???

Reply:
I think there is no use case for this migration. I don't know who could be affected, and if they are it's a test system.